test/celeritas/global/AlongStep.test.cc

```diff
@@ -87,6 +91,52 @@ TEST_F(KnAlongStepTest, basic)
     }
 }
 
+TEST_F(MockAlongStepTest, basic)
+{
+    size_type num_tracks = 10;
+    Input inp;
+    inp.particle_id = this->particle()->find("celeriton");
+    {
+        inp.energy = MevEnergy{1};
+        auto result = this->run(inp, num_tracks);
+        EXPECT_SOFT_EQ(0.29312, result.eloss);
+        EXPECT_SOFT_EQ(0.48853333333333, result.displacement);
+        EXPECT_SOFT_EQ(1, result.angle);
+        EXPECT_SOFT_EQ(1.881667426791e-11, result.time);
+        EXPECT_SOFT_EQ(0.48853333333333, result.step);
+        EXPECT_EQ("eloss-range", result.action);
+    }
+    {
+        inp.energy = MevEnergy{1e-6};
+        auto result = this->run(inp, num_tracks);
+        EXPECT_SOFT_EQ(1e-06, result.eloss);
+        EXPECT_SOFT_EQ(5.2704627669473e-05, result.displacement);
+        EXPECT_SOFT_EQ(1, result.angle);
+        EXPECT_SOFT_EQ(1.2431209185653e-12, result.time);
+        EXPECT_SOFT_EQ(5.2704627669473e-05, result.step);
+        EXPECT_EQ("physics-discrete-select", result.action);
+    }
+    {
+        inp.energy = MevEnergy{1e-12};
```

Comment:
They're showing up in real life because we're not applying the energy loss cut *after* energy loss. I've got an incoming PR that will fix that in the usual cases, but because it doesn't apply on the boundaries there's still a microscopic chance that we'll have a very low energy particle. (Or maybe we'll get some other roundoff issue from a post-collision.) In any case it's good to make sure the code behaves in those cases...